test/SlowTests/Utils/RavenDocumentsByEntityName.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Abstractions.Indexing;
+using Raven.Client.Indexes;
+using Raven.Client.Indexing;
+
+namespace SlowTests.Utils
+{
+    ///<summary>
+    /// Create an index that allows to tag entities by their entity name
+    ///</summary>
+    public class RavenDocumentsByEntityName : AbstractIndexCreationTask
```

Comment:
Actually, this was a part of the core code in the 3.5. 
I moved it to separated file because it is been used in a test.